src/vcpkg/binarycaching.cpp

```diff
@@ -721,56 +731,45 @@ namespace
                 cmdlines.push_back(std::move(cmdline));
             }
 
-            const size_t current_restored = m_restored.size();
-
+            const size_t total_restore_attempts = nuget_refs.size();
             for (const auto& cmdline : cmdlines)
             {
                 if (nuget_refs.empty())
                 {
                     break;
                 }
 
-                [&] {
-                    generate_packages_config();
-                    run_nuget_commandline(cmdline);
-                }();
-
-                Util::erase_remove_if(nuget_refs, [&](const std::pair<PackageSpec, NugetReference>& nuget_ref) -> bool {
+                generate_packages_config(fs, packages_config, nuget_refs);
+                run_nuget_commandline(cmdline);
+                Util::erase_remove_if(nuget_refs, [&](const NuGetPrefetchAttempt& nuget_ref) -> bool {
                     auto nupkg_path =
-                        paths.package_dir(nuget_ref.first) / vcpkg::u8path(nuget_ref.second.id + ".nupkg");
+                        paths.package_dir(nuget_ref.spec) / vcpkg::u8path(nuget_ref.reference.id + ".nupkg");
```

Comment:
This should go into buildtrees

Reply:
I believe that this is due to the operation of `nuget.exe`, not a choice on the part of vcpkg.

Reply:
```
                    .string_arg("-OutputDirectory")
                    .path_arg(paths.packages)
```

isn't where it comes from?

Reply:
Oh I see, the client leaves the .nupkg turd around after extraction. Adding comment.